_posts/2016-11-09-making-a-small-object.md

```diff
@@ -0,0 +1,144 @@
+---
+layout: post
+title: Making a Small Object
+excerpt: The thought process behind creating a simple object to help deal with complex ActiveRecord ORDER BY queries with a strong focus on the Single Responsibility Principal.
+published: true
+categories: SQL Ruby ActiveRecord
+---
+
+I get excited about small objects when I program. I often open up files that are hundreds of lines long with mixed responsibilities and lots of complexity.
+I recently had an opportunity to make a small object with a single responsibility. The object would be used to build the SQL fragment that gets passed as a parameter to ActiveRecord's `.order()` method.
```

Comment:
You could provide a little more background. Something like: 

`Recently, I had to add functionality to a behemoth object which had W lines of code, X methods, and about Y different potential code paths. Rather than tacking on an additional Z lines of code and adding yet another responsibility to this large object, I turned the task into an opportunity to make a small object with a single responsibility.`